scripts/modules/b2b-account/email-quote.js

```diff
@@ -0,0 +1,110 @@
+define([
+    "modules/jquery-mozu",
+    'modules/api',
+    "underscore",
+    "hyprlive",
+    "modules/backbone-mozu",
+    "hyprlivecontext",
+    "modules/models-customer",
+    'modules/mozu-grid/mozugrid-view',
+    'modules/mozu-grid/mozugrid-pagedCollection',
+    "modules/views-paging",
+    'modules/editable-view',
+    "modules/models-quotes",
+    "modules/models-b2bcontacts"],
+    function ($, api, _, Hypr, Backbone, HyprLiveContext, CustomerModels, MozuGrid, MozuGridCollection,
+        PagingViews, EditableView, QuoteModels, B2bModels) {
+
+       
+        var accountInfo=require.mozuData('user');
+        var invalidEmails=[];
+        var validEmails=[];
+
+        var emailQuoteView = Backbone.MozuView.extend({
+            templateName: "modules/b2b-account/quotes/email-quote",
+
+            initialize: function () {
+                Backbone.MozuView.prototype.initialize.apply(this, arguments);
+            },
+            render: function (accountId) {
+                var self = this;
+                self.model.set("accountId",accountId);
+                Backbone.MozuView.prototype.render.apply(this, arguments);
+                if(!accountInfo.isSalesRep)
+                $("#sendEmailText").val(accountInfo.email);
+                $(".modal-dialog").attr("id", "mz-email-modal");
+            $('[data-mz-action="handleSendemail"]').click(function() {
+               var emails; 
+               if(!self.model.attributes.inputEmails && !accountInfo.isSalesRep){
+               emails=$("#sendEmailText").val();
+               }
+               else{
+               emails=self.model.attributes.inputEmails;
+               }
+                var result = emails.split(",");
+                for(var i = 0;i < result.length && i < 20;i++){
+                    if(!result[i].match(Backbone.Validation.patterns.email)){
+                        invalidEmails.push(result[i]);
+                    }else{
+                        validEmails.push(result[i]);
+                    }
+                }
+                if(invalidEmails.length===0){
+                var quote = new QuoteModels.Quote({
+                    id: self.model.get("accountId")
+                });
+                return quote.apiModel.email().then(function (res) {
+                   // to do after email sent successfully
+                    // emailAddress:validEmails
+                }, function (error) {
+                    if (error !== undefined){self.showMessageBar(error.message);}else{
+                        self.showMessageBar("Request Error");
+                    } 
+                }
+                );
+            }else{
+                for(var j = 0;j < invalidEmails.length;j++){
+                    self.showMessageBar(invalidEmails[j]);
+                }
+                if(invalidEmails.length ==  1){self.showMessageBar("Invalid Email Address ");} else{
```

Comment:
Why we need this message again, we are already looped over invalidEmails[] and showed error messages. Why we need this if/else again. Show, only one error messages with all invalid emails in that message. with a proper readable message.